daemon/daemon_test.go

```diff
@@ -313,3 +327,163 @@ func TestFindNetworkErrorType(t *testing.T) {
 		t.Error("The FindNetwork method MUST always return an error that implements the NotFound interface and is ErrNoSuchNetwork")
 	}
 }
+
+//
+// https://github.com/moby/moby/issues/38386
+//
```

Comment:
Wondering if this comment should go as GoDoc for `TestRegistryHosts`. Perhaps also summarising what its testing for (which may include this URL to refer for further details).

Having a GoDoc on the test may help "future visitors" of the code to get more context on what's being tested, and to verify the test is still testing for purpose it was written for (I encountered various cases where the original _intent_ got lost in the weeds when tests moved or had to be updated/refactored).